public/scripts/pubminer.js

```diff
@@ -0,0 +1,115 @@
+                $.get('/detail/' + uidValue, null, function(response) {
+                    if (response.error) {
+                        expansionBody.text(response.error);
+                    } else {
+                        expansionBody.html("");
+                        for (var key in response) {
+                            if (response.hasOwnProperty(key)) {
+                                expansionBody.append($("<strong>", {html: key}));
+                                expansionBody.append(" " + response[key]);
+                                expansionBody.append("<br/>")
+                            }
+                        }
+                    }
+                });
+            }
+        } else {
+        }
+    });
+};
+
+// close the expanded row
+// code adapted from samples on PatternFly.org
+$.fn.addCloseRowHandler = function() {
+    this.on("click", function (){
+        $(this).parent().addClass("hidden")
+        .parent().removeClass("list-view-pf-expand-active")
+        .find(".fa-angle-right").removeClass("fa-angle-down");
+    });
+};
+
+// add styling to rows that have been check-selected
+// code adapted from samples on PatternFly.org
+$.fn.addRowCheckboxHandler = function() {
+    this.change(function (e) {
+        if ($(this).is(":checked")) {
+            $(this).closest('.list-group-item').addClass("active");
+        } else {
+            $(this).closest('.list-group-item').removeClass("active");
+        }
+    });
+};
+
+$(document).ready(function() {
+
+    // the next three function calls add handlers to the rows initially displayed.
+
+    // row checkbox selection
+    $("input[type='checkbox']").addRowCheckboxHandler();
+
+    // click the list-view heading then expand a row
+    $(".list-group-item-header").addArrowClickHandler();
+
+    // click the close button, hide the expand row and remove the active status
+    $(".list-group-item-container .close").addCloseRowHandler();
+
+    // handle new rows added to display
+    $("#moreButton").click(function() {
+
+        let params = { start: $("#nextRow").val(),
+                       count: $("#rowCount").val()};
+
+        // make sure there are more rows more to get
+        if (params.start != 0) {
+
+            $.get('/results/' + $("#webenv").val() + '/' + $("#querykey").val(),
+                params, function(response) {
+
+                    let newRows = $.parseHTML(response)
+
+                    // add handlers to the new rows
+
+                    // row checkbox selection
+                    $(newRows).find("input[type='checkbox']").addRowCheckboxHandler();
+
+                    // click the list-view heading then expand a row
+                    $(newRows).find(".list-group-item-header").addArrowClickHandler();
+
+                    // click the close button, hide the expand row and remove the active status
+                    $(newRows).find(".list-group-item-container .close").addCloseRowHandler();
```

Comment:
minor: it might be worth considering if there is a way to centralize the handler adding logic that's executed for document load and adding new rows since they are similar (but it's a minor consideration).

Reply:
It's a good point, and brings to mind a question I had.  At some place in the code we need to make the initial search and return the first set of results.  This used to be in search.js, but it clearly couldn't stay there.  Right now it is at the controller (route) level. When the initial search request comes in to the back and, the route code makes two calls.  Is this the right place, or should we move this even further back to the JS in the UI? If so, then that would take care of the centralizing the handler code.